app/decorators/module_overview_decorator.rb

```diff
@@ -55,98 +50,76 @@ def module_completed_at
 
     # providing the next page name enables the hyperlink
-    if clickable?(topic_item: topic_item, submodule: submodule)
-      next_item_name = topic_status.eql?(:started) ? resume_page : topic_item
+    if clickable?(subsection_item: subsection_item, submodule: submodule)
+      furthest_topic_page_name = subsection_status.eql?(:started) ? resume_page : subsection_item
     end
 
     # SummativeAssessmentProgress conditional pass
     if failed_attempt?
-      topic_status = :failed if topic_item.assessment_intro?
-      next_item_name = nil if topic_item.confidence_intro?
+      subsection_status = :failed if subsection_item.assessment_intro?
+      furthest_topic_page_name = nil if subsection_item.confidence_intro?
     elsif successful_attempt?
-      topic_status = :completed if topic_item.assessment_intro?
-      next_item_name = topic_item if topic_item.confidence_intro?
+      subsection_status = :completed if subsection_item.assessment_intro?
+      furthest_topic_page_name = subsection_item if subsection_item.confidence_intro?
     end
 
     [
-      topic_item.training_module,   # TrainingModule [mod]
-      topic_item.model.heading,     # String (content page heading) [topic_heading]
-      next_item_name,               # String/nil (module_item name) [next_item]
-      topic_status,                 # Symbol (all items viewed) [status]
+      subsection_item.training_module,   # TrainingModule [mod]
+      subsection_item.model.heading,     # String (content page heading) [subsection_heading]
+      furthest_topic_page_name,          # String/nil (module_item name) [next_item]
+      subsection_status,                 # Symbol (all items viewed) [status]
     ]
   end
 
   # @param submodule [String]
-  # @param topic_item [ModuleItem]
+  # @param subsection_item [ModuleItem]
   #
   # @return [Boolean]
-  def clickable?(submodule:, topic_item:)
+  def clickable?(submodule:, subsection_item:)
+    return all?([subsection_item]) if submodule.nil?
+
     submodule_intro = mod.module_items_by_submodule(submodule).first
     return false unless visited?(submodule_intro)
 
-    current_topic = find_current_topic(topic_item.topic_name, submodule)
+    current_topic = find_topic(subsection_item.topic_name, submodule)
     current_topic_items = current_topic.values.first.to_a
 
-    previous_submodule = find_previous_submodule(topic_item.submodule_name)
-    previous_submodule_items = previous_submodule.values.first.to_a
-
-    if current_topic && previous_submodule
-      all?(current_topic_items) && all?(previous_submodule_items)
-    else
-      all?([topic_item])
-    end
-  end
-
-  # @param num [String]
-  #
-  # @return [String]
-  def previous(num)
-    (num.to_i - 1).to_s
-  end
-
-  # @param num [String]
-  #
-  # @return [Array<Array>] ['1'] [item, item, item]
-  def find_previous_submodule(num)
-    mod.items_by_submodule.select do |sub_num, _|
-      previous(num).eql?(sub_num)
+    # if current_topic && previous_submodule
```

Comment:
Still to tidy up? It'd be great if we no longer need this complexity but I still can't remember exactly why it is there. Is there a test for it?